src/embind/embind.js

```diff
@@ -584,6 +589,46 @@ var LibraryEmbind = {
     });
   },
 
+#if WASM_BIGINT
+  _embind_register_bigint__deps: [
+    'embind_repr', '$readLatin1String', '$registerType', '$integerReadValueFromPointer'],
+  _embind_register_bigint: function(primitiveType, name, size, minRange, maxRange) {
+    name = readLatin1String(name);
+
+    var shift = getShiftFromSize(size);
+
+    var isUnsignedType = (name.indexOf('u') != -1);
+
+    // maxRange comes through as -1 for uint64_t (see issue 13902). Work around that temporarily
+    if (isUnsignedType) {
+      maxRange = 18446744073709551615n;
+    }
+
+    registerType(primitiveType, {
+        name: name,
+        'fromWireType': function (value) {
+          return value
+        },
+        'toWireType': function (destructors, value) {
+          if (typeof value === "number" || typeof value === "boolean") {
+            value = BigInt(value);
```

Comment:
Does it make sense to do these conversions for the caller?

Reply:
Good question. I would say no - we shouldn't try to be "friendlier" than JS does, and JS does not do these conversions itself. But I don't feel strongly.

Reply:
I agree, I'd rather make things stricter - if some implicit coercions turn out to be absolutely necessary, we can add them later, but usually they cause more problems than solve.

Reply:
It feels right to remove the implicit coercions. Thanks for the input.